lib/Elastica/Transport/Guzzle.php

```diff
@@ -156,7 +175,12 @@ protected function _getBaseUrl(Connection $connection)
         if (!empty($url)) {
             $baseUri = $url;
         } else {
-            $baseUri = $this->_scheme.'://'.$connection->getHost().':'.$connection->getPort().'/'.$connection->getPath();
+            $baseUri = (string) Uri::fromParts([
+                'scheme' => $this->_scheme,
+                'host' => $connection->getHost(),
+                'port' => $connection->getPort(),
+                'path' => ltrim('/', $connection->getPath()),
```

Comment:
I didn't try the guzzle transport yet. Was just looking at it out of interest. But I assume the `ltrim` should be removed.